lib/modules/datasource/terraform-module/index.ts

```diff
@@ -159,6 +160,12 @@ export class TerraformModuleDatasource extends TerraformDatasource {
         version,
       })),
     };
+
+    // Add the source URL if given
+    if (res.modules[0].source && validateUrl(res.modules[0].source)) {
```

Comment:
```suggestion
    if (validateUrl(res.modules[0].source)) {
```
With this PR validateUrl will do this for you.
https://github.com/renovatebot/renovate/pull/25014